### DISCUSSION TITLE
Quick question regarding InitializeComponent() method

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#7561] [Q&A] [answered]
Creating a new window or control gives a template with an `InitializeComponent()` method which is called by the constructor.

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
        }

The only reason I can think for having this is the scenario where there are multiple constructors.

What's the thinking for having this method? Is there a reason to keep it in my classes rather than do all the initialisation in the constructor itself including the Load().

**Comment by maxkatz6** [comment, marked as answer]
You can call it in constructor. InitializeComponent is more a match to the WPF behavior and IIRC it was required in Avalonia before.
Currently you can get rid of it completely with source generator https://github.com/AvaloniaUI/Avalonia.NameGenerator
